### ISSUE TITLE
DateTimeExtensions.Truncate does not work correctly with DateTimePeriod.Week

### ISSUE THREAD

**Issue by dermeister0** [Saritasa/SaritasaTools#54]
```
using Saritasa.Tools.Common.Extensions;
using Saritasa.Tools.Common.Utils;
using System;
using System.Globalization;
using Xunit;

namespace Test071100
{
    public class DateTimeExtensionsTest
    {
        [Fact]
        public void Truncate_USCulture_ReturnsJuly8()
        {
            var date = new DateTime(2018, 7, 14, 0, 0, 0, DateTimeKind.Utc);
            var cultureInfo = new CultureInfo("us");
            var result = date.Truncate(DateTimePeriod.Week, cultureInfo);
            Assert.Equal(8, result.Day); // Actual: 8
        }

        [Fact]
        public void Truncate_RussianCulture_ReturnsJuly9()
        {
            var date = new DateTime(2018, 7, 15, 0, 0, 0, DateTimeKind.Utc);
            var cultureInfo = new CultureInfo("ru");
            var result = date.Truncate(DateTimePeriod.Week, cultureInfo);
            Assert.Equal(9, result.Day); // Actual: 16
        }

        [Fact]
        public void Truncate_CustomCulture_ReturnsJuly11()
        {
            var date = new DateTime(2018, 7, 17, 0, 0, 0, DateTimeKind.Utc);
            var cultureInfo = new CultureInfo("us");
            cultureInfo.DateTimeFormat.FirstDayOfWeek = DayOfWeek.Wednesday;
            var result = date.Truncate(DateTimePeriod.Week, cultureInfo);
            Assert.Equal(11, result.Day); // Actual: 18
        }
    }
}
```

**Comment by dermeister0** [comment]
See possible calculations:
https://stackoverflow.com/questions/7703410/how-can-i-change-which-day-is-the-first-day-of-the-week

**Comment by krasninja** [comment]
`var date = new DateTime(2018, 7, 15, 0, 0, 0, DateTimeKind.Utc);` In US it is the actual beginning of week (15), isn't it?

**Comment by dermeister0** [comment]
You're right. I modified the example.

**Comment by krasninja** [comment]
Fixed in 0.4.8 .
